src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -9,22 +9,25 @@ namespace Microsoft.CodeAnalysis.CSharp
     internal sealed class BestTypeInferrer
     {
         private readonly ConversionsBase _conversions;
-        private BestTypeInferrer(ConversionsBase conversions)
+        private readonly Compilation _compilation;
+
+        private BestTypeInferrer(ConversionsBase conversions, Compilation compilation)
```

Comment:
It looks like we're always passing a `Conversions` instance rather than `ConversionsBase`. Would it make sense to change `InferBestType` methods to take a `Conversions` and use `Conversions.Compilation` here instead?